src/tests.c

```diff
@@ -0,0 +1,287 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+
+#include "./verify.h"
+
+#ifndef BENCH_ITER
+#define BENCH_ITER 100
+#endif
+
+void simple_test(int *tests, int *pass, int *fail, VF_return_t expected,
+                 uint8_t *pubkey, int pubkey_l, uint8_t *document,
+                 int document_l, uint8_t *signature, int signature_l,
+                 char *msg) {
+
+  struct timeval start;
+  struct timeval end;
+
+  struct Error *err = NULL;
+
+  gettimeofday(&start, NULL);
```

Comment:
Cool!  thanks for the pointer to that function.  For the first version, gettimeofday is probably good enough, especially since we're not doing any thread switching in these unit tests.

Reply:
Notice I am talking about thread system in a system level. Say, in the middle of the verify function, the kernel switches to a kernel thread responsible to flush system memory pages, `gettimeofday` will account for that time, while `clock_gettime` won't.